src/k2/module/k23si/Module.h

```diff
@@ -146,6 +149,28 @@ class K23SIPartitionModule {
     // return true if request is valid
     bool _validateStaleWrite(dto::K23SIWriteRequest& request, std::deque<dto::DataRecord>& versions);
 
+    template <class RequestT>
+    Status _validateReadRequest(const RequestT& request) const {
+        if (!_validateRequestPartition(request)) {
+            // tell client their collection partition is gone
+            return dto::K23SIStatus::RefreshCollection("collection refresh needed in read-type request");
+        }
+        if (!_validateRequestParameter(request)){
```

Comment:
The name for this func is too vague. Is it just supposed to check for the existence of a partition key? if so, let's call it that